test/rules/always_put_control_body_on_new_line.dart

```diff
@@ -19,6 +19,13 @@ testIfElse() {
     return; // OK
   else return; // LINT
 
+  if (false) {
+  }
+  else if (false) { // OK
```

Comment:
Actually this lint rule does conflict with `dartfmt` because the Flutter style guide disagree on some formatting rules. So this lint can be seen as an helper until the flutter formatter lands. And once landed I guess this lint could be removed.

BTW I'm a little worried about the existence of 2 formatters.

And last thing, I haven't merge rights :)